app/src/main/java/org/mozilla/fenix/settings/address/ext/Address.kt

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address.ext
+
+import mozilla.components.concept.storage.Address
+
+/**
+ * Generate an [AddressItemText] from a given [Address]. A well-formed [Address] will have
+ * a full name in the label and a street address as a description. If either are not present,
+ * the label and description will be based off lower priority properties.
+ */
+fun Address.toItemText(): AddressItemText {
+    val label = getName().ifEmpty { iterateFields("") }
+    val description = iterateFields(label)
+    return AddressItemText(label, description)
+}
+
+/**
+ * Holds item display values about an [Address].
+ *
+ * @property label The label for the [Address].
+ * @property description The subtitled description for an [Address].
+ */
+data class AddressItemText(
+    val label: String,
+    val description: String
+)
+
+private fun Address.getName(): String {
+    val middleName = if (additionalName.isEmpty()) "" else "$additionalName "
+    return if (givenName.isNotEmpty()) "$givenName $middleName$familyName" else familyName
```

Comment:
Just doing a quick scan but are we missing a space between `$middleName$familyName`?

Reply:
Oh nvm, I see the space above